internal/expr/query.go

```diff
@@ -81,15 +94,34 @@ func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
 		switch tm := typeMember.(type) {
 		case *structField:
 			val = v.Field(tm.index)
+			qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(argCount), val.Interface()))
+			argCount++
 		case *mapKey:
 			val = v.MapIndex(reflect.ValueOf(tm.name))
 			if val.Kind() == reflect.Invalid {
 				return nil, fmt.Errorf(`map %q does not contain key %q`, outerType.Name(), tm.name)
 			}
+			qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(argCount), val.Interface()))
+			argCount++
+		case *sliceType:
+			sliceLen := v.Len()
+			if sliceLen == 0 {
+				return nil, fmt.Errorf(`slice arg with type %q has length 0`, tm.sliceType.Name())
+			}
+			sliceLens = append(sliceLens, sliceLen)
+			i := 0
+			for i = 0; i < v.Len(); i++ {
+				sv := v.Index(i)
+				qargs = append(qargs, sql.Named("sqlair_"+strconv.Itoa(argCount), sv.Interface()))
+				argCount++
+			}
+		default:
+			return nil, fmt.Errorf(`internal error: unknown type: %T`, tm)
```

Comment:
We are using `panic` for the internal error above, should we do the same here?

Reply:
We only use panic in the `sql` method because it doesn't make sense for it to return an error (i.e. the only errors it could return are internal errors). Here we are already returning an error so we might as well make use of it.